src/components/Contacts/ContactListTableMobile.jsx

```diff
@@ -0,0 +1,214 @@
+// React Imports
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import SendIcon from '@mui/icons-material/Send';
+import Typography from '@mui/material/Typography';
+import { useTheme } from '@mui/material/styles';
+// Custom Hooks Imports
+import useNotification from '@hooks/useNotification';
+// Util Imports
+import { saveToClipboard } from '@utils';
+// Component Imports
+import ContactProfileIcon from './ContactProfileIcon';
+
+/**
+ * @typedef {object} Contact
+ * @property {string} webId - The Web ID of the contact
+ * @property {string} givenName - The given name of the contact
+ * @property {string} familyName - The family name of the contact
+ */
+
+/**
+ * ContactListTableMobile - Component for displaying contacts in a DataGrid
+ *
+ * @memberof Contacts
+ * @name ContactListTableMobile
+ * @param {object} Props - The props for ContactListTableMobile
+ * @param {Contact[]} Props.contacts - The list of contacts to display
+ * @param {Function} Props.deleteContact - Function to delete a contact
+ * @param {Function} Props.handleSendMessage - Function to handle sending a message
+ * @returns {React.JSX.Element} The ContactListTableMobile component
+ */
+const ContactListTableMobile = ({ contacts, deleteContact, handleSendMessage }) => {
+  const { addNotification } = useNotification();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+  const navigate = useNavigate();
+  const theme = useTheme();
+
+  const handleClick = (event, contact) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(contact.webId);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  const handleMenuItemClick = (action, contact) => () => {
+    action(contact);
+    handleClose();
+  };
+
+  const handleProfileClick = (contact) => {
+    // TODO: Consider abstracting this into a reusable function
+    navigate(`/contacts/${encodeURIComponent(contact.webId)}`, {
+      state: { contact }
```

Comment:
This itself won't work. You're still missing the handler function which help sets up the correct profile.

Reply:
You need to use the `setContact` from `DocumentListContext` and create that new handler for the mobile view as well as importing `getWebIdDataset` to check if the webId is valid.

![Screenshot 2024-07-03 at 12 54 34 PM](https://github.com/codeforpdx/PASS/assets/14917816/9040a857-b1e7-47b2-b7cc-200c4e0caea9)